File: first_raspberry_test.c
#include <wiringPi.h>
#include <stdio.h>
#include <time.h>


int main(void){
	const int coinGpio = 15;
	const int aGpio = 0;
	const int bGpio = 1;
	const int cGpio = 16;
	wiringPiSetup();
	pinMode(coinGpio, INPUT);
	pinMode(aGpio, INPUT);	
	pinMode(bGpio, INPUT);
	pinMode(cGpio, INPUT);
	time_t start = time(NULL);
	int coin = 0;
	int lastState = digitalRead(gpio);
	for(time_t cur = start; cur - start < 20; cur = time(NULL))
	{	
		int curState = digitalRead(gpio);
		
		if(curState == 1 && lastState == 0)
		{
			coin++;
			printf("[%d] Coincidence, Time: %d \n", coin, (int)(cur-start));
		}
		lastState = curState;
	}
	printf("Coincidences: %i \n",coin);
}

File: first_raspberry_test.py
import RPi.GPIO as gpio
import time

coinGPIO = 14
gpio.setmode(gpio.BCM)
gpio.setup(coinGPIO, gpio.IN)

start = time.time()
cur = time.time()
coin = 0

while cur-start < 30:
	#if round(cur-0.00001,0)!=round(cur,0):
		#print("Time: " + str(round(cur-start,2)) + "s, Coincidences: " + str(coin))
	if gpio.input(coinGPIO):
		coin += 1
	cur = time.time()
print("Coincidences found: " + str(coin))
gpio.cleanup()
